### ISSUE TITLE
This regex may be stucked by input

### ISSUE THREAD

**Issue by 2bdenny** [ajaxorg/ace#3638]
https://github.com/ajaxorg/ace/blob/1f8333bbd57f293d9a5dc5e9ea07a9381963f4ad/lib/ace/tokenizer.js#L179

Hello! I have tested all regexes in the project, and find this regex can be stucked by some input. For example, the following code would cost over 30 seconds. =)

```
var str = String.raw`[[@[[@[@[_\>\?\|)\\>\?\|)\\>\?\|)\\>\?\|)\\>\?\|)\\>\?\|)\\>\?\|)\>\?\|\x<,T'@[x<,T'@[_`
var r = str.replace(
            /\[(?:\\.|[^\]])*?\]|\\.|\(\?[:=!]|(\()/g,
            function(x, y) {return y ? "(?:" : x;}
        );
```

**Comment by nightwing** [comment]
Hi
how did you test all regexps? Did you use a tool to test regexps automatically?

**Comment by 2bdenny** [comment]
Yes, the tool is under development and is not yet stable (Orz). It will be open source on GitHub later.
